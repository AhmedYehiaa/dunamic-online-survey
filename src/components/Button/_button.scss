.button {
  padding: $s-size;
  border: none;
  border-radius: $xs-size;
  color: $white;
  font-weight: bold;
  font-size: $m-size;
  transition: all 300ms;
  cursor: pointer;
  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }
  &:focus {
    outline:0;
  }
}

.btn-large {
  padding: $l-size;
  margin-bottom: $xl-size;
  max-width: 350px; 
  width: 100%;
  border: none;
  font-size: 25px;
  text-transform: uppercase;
  color: #ffffff;
  font-weight: 700;
}

.btn-medium {
  padding: $m-size;
  height: 55px;
  width: 100px;
}

.btn-primary {
  background: #2196F3;
  &:hover {
    box-shadow: 0px 0px 20px 4px #2196f380;
  }
}

.btn-success {
  background-color: $success;
}

.btn-warning {
  background-color: $warning;
}

.btn-danger {
  background-color: $danger;
}

.btn-back {
  background-color: $gray;
}
