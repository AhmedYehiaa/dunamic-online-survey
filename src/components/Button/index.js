/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

export const TYPES = {
  PRIMARY: 'btn-primary',
  WARNING: 'btn-warning',
  DANGER: 'btn-danger',
  SUCCESS: 'btn-success',
  BACK: 'btn-back',
  NEXT: 'btn-next',
};

export const SIZES = {
  SMALL: 'btn-small',
  MEDIUM: 'btn-medium',
  LARGE: 'btn-large',
};

const Button = ({
  type, text, disabled, buttonType, buttonSize, onClick,
}) => (
  <button
    type={type}
    className={`button ${buttonType} ${buttonSize}`}
    disabled={disabled}
    onClick={onClick}
  >
    {text}
  </button>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  buttonType: PropTypes.string,
  buttonSize: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  buttonType: TYPES.PRIMARY,
  buttonSize: SIZES.MEDIUM,
  disabled: false,
};

export default Button;
