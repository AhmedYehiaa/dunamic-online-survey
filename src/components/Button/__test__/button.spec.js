import React from 'react';
import { shallow } from 'enzyme';
import Button, { TYPES, SIZES } from '../index';

it('it should renders the button', () => {
  const onClickNext = jest.fn();
  const disableNextButton = false;

  const wrapper = shallow(<Button
    type="button"
    onClick={onClickNext}
    text="Next"
    disabled={disableNextButton}
    buttonType={TYPES.PRIMARY}
    buttonSize={SIZES.MEDIUM}
  />);
  expect(wrapper).toMatchSnapshot();
});
