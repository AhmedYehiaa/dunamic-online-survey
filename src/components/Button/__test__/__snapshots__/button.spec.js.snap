// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders the button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonSize="btn-medium"
    buttonType="btn-primary"
    disabled={false}
    onClick={[MockFunction]}
    text="Next"
    type="button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Next",
      "className": "button btn-primary btn-medium",
      "disabled": false,
      "onClick": [MockFunction],
      "type": "button",
    },
    "ref": null,
    "rendered": "Next",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Next",
        "className": "button btn-primary btn-medium",
        "disabled": false,
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": "Next",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
